lib/fizzy/execute.cpp

```diff
@@ -415,15 +420,15 @@ inline bool store_into_memory(bytes& memory, Stack<uint64_t>& stack, const uint8
 }
 
 template <typename Op>
-inline void unary_op(Stack<uint64_t>& stack, Op op) noexcept
+inline void unary_op(OperandStack& stack, Op op) noexcept
 {
     using T = decltype(op(stack.pop()));
     const auto a = static_cast<T>(stack.pop());
```

Comment:
> Not easy with of the casts present there.

Not sure why, at least for `unary_op` here this seems possible:
```suggestion
    stack.top() = op(static_cast<T>(stack.top()))
```
or 
```suggestion
    auto& a = stack.top();
    a = op(static_cast<T>(a))
```